dipy/align/imaffine.py

```diff
@@ -0,0 +1,1010 @@
+
+        By default, the transformed image is sampled at a grid defined by
+        `self.codomain_shape` and `self.codomain_grid2world`. If such
+        information was not provided then `sampling_grid_shape` is mandatory.
+
+        Parameters
+        ----------
+        image : array, shape (X, Y) or (X, Y, Z)
+            the image to be transformed
+        interp : string, either 'linear' or 'nearest'
+            the type of interpolation to be used, either 'linear'
+            (for k-linear interpolation) or 'nearest' for nearest neighbor
+        image_grid2world : array, shape (dim + 1, dim + 1), optional
+            the grid-to-world transform associated with `image`.
+            If None (the default), then the grid-to-world transform is assumed
+            to be the identity.
+        sampling_grid_shape : sequence, shape (dim,), optional
+            the shape of the grid where the transformed image must be sampled.
+            If None (the default), then `self.codomain_shape` is used instead
+            (which must have been set at initialization, otherwise an exception
+            will be raised).
+        sampling_grid2world : array, shape (dim + 1, dim + 1), optional
+            the grid-to-world transform associated with the sampling grid
+            (specified by `sampling_grid_shape`, or by default
+            `self.codomain_shape`). If None (the default), then the
+            grid-to-world transform is assumed to be the identity.
+        resample_only : Boolean, optional
+            If False (the default) the affine transform is applied normally.
+            If True, then the affine transform is not applied, and the input
+            image is just re-sampled on the domain grid of this transform.
+        Returns
+        -------
+        transformed : array, shape `sampling_grid_shape` or
+                      `self.codomain_shape`
+            the transformed image, sampled at the requested grid
+        """
+        transformed = self._apply_transform(image, interp, image_grid2world,
+                                            sampling_grid_shape,
+                                            sampling_grid2world,
+                                            resample_only,
+                                            apply_inverse=True)
+        return np.array(transformed)
+
+
+class MattesMIMetric(MattesBase):
+    def __init__(self, nbins=32, sampling_proportion=None):
+        r""" Initializes an instance of the Mattes MI metric
+
+        This class implements the methods required by Optimizer to drive the
+        registration process by making calls to the low level methods defined
+        in MattesBase.
+
+        Parameters
+        ----------
+        nbins : int, optional
+            the number of bins to be used for computing the intensity
+            histograms. The default is 32.
+        sampling_proportion : float in (0,1], optional
```

Comment:
`float in` is a bit confusing because it reads as Python `in`, rather than math `in`.  Also you need to note that value can be None.  Maybe

``None or float in interval (0, 1], optional``